packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2431,13 +2439,60 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     updateKeepAlive();
   }
 
+  String _cachedText = '';
+  Rect? _cachedFirstRect;
+  Size _cachedSize = Size.zero;
+  int _cachedPlaceholder = -1;
+
+  void _updateSelectionRects({bool force = false}) {
+    if (defaultTargetPlatform != TargetPlatform.iOS)
+      return;
+    // This is to avoid sending selection rects on non-iPad devices
```

Comment:
Period at the end.

Reply:
I guess we have no better way to determine if this is iPadOS...  At least we should make `1536.0` a constant called `_kIPadWidth` or something like that.